spyder/utils/programs.py

```diff
@@ -817,84 +830,67 @@ def is_module_installed(module_name, version=None, installed_version=None,
     If version is not None, checking module version
     (module must have an attribute named '__version__')
 
-    version may starts with =, >=, > or < to specify the exact requirement ;
+    version may start with =, >=, > or < to specify the exact requirement ;
     multiple conditions may be separated by ';' (e.g. '>=0.13;<1.0')
 
     interpreter: check if a module is installed with a given version
     in a determined interpreter
     """
-    if interpreter:
-        if is_python_interpreter(interpreter):
-            checkver = inspect.getsource(check_version)
-            get_modver = inspect.getsource(get_module_version)
-            stable_ver = inspect.getsource(is_stable_version)
-            ismod_inst = inspect.getsource(is_module_installed)
+    if installed_version is not None:
+        '''
+        installed_version only passed in from dependencies.Dependency.check
+        which retrieved it from get_module_version, so we know the module is
+        already installed
+        '''
+        return True
 
-            f = tempfile.NamedTemporaryFile('wt', suffix='.py',
-                                            dir=get_temp_dir(), delete=False)
+    module_version = None
+    if interpreter is not None:
+        if is_python_interpreter(interpreter):
+            cmd = 'import %s; print(%s.__version__)' % ((module_name,) * 2)

```

Comment:
This is incorrect because not all modules have a `__version__` variable, for instance `keyring`. Therefore, this will report `keyring` as missing when in fact it's installed.

The previous version reported it as installed, by the way.

Reply:
So I think you need to run two `run_program` calls: one to detect if the module is present (simply by trying to import it); and the second trying to get its version (which should be assigned as `None` if that call returns an error).

Reply:
Okay, I'll look at this.

Reply:
Done

Reply:
I did not need to ``run_program`` twice, but just improved ``cmd``